src/Apps/CollectorVerifcation/routes.tsx

```diff
@@ -0,0 +1,42 @@
+import { ErrorPage } from "Components/ErrorPage"
+import { RouteConfig } from "found"
+import React from "react"
+import { graphql } from "react-relay"
+import createLogger from "Utils/logger"
+import { CollectorVerificationApp } from "./CollectorVerificationApp"
+
+const logger = createLogger("Apps/CollectorVerification/routes")
+
+export const routes: RouteConfig[] = [
+  {
+    path: "/collector-verification/:id",
+    Component: CollectorVerificationApp,
+    query: graphql`
+      query routes_CollectorVerificationAppQuery($id: String!) {
+        me {
+          ...CollectorVerificationApp_me @arguments(id: $id)
+          internalID
+          identityVerification(id: $id) {
+            userID
+          }
+        }
+      }
+    `,
+    render: ({ Component, props }) => {
+      logger.log("render")
+      if (Component && props) {
+        const { me } = props as any
+        const { identityVerification } = me
+
+        if (
+          !identityVerification ||
+          identityVerification.userID !== me.internalID
+        ) {
+          return <ErrorPage code={404} />
+        }
+
+        return <Component {...props} />
+      }
```

Comment:
Using custom render methods like this breaks the routers internal functionality slightly. 

All of these guards can move into `CollectorVerificationApp` (exit early if 404), and this `render` function can be removed.

Reply:
ah, i think @yuki24 taught our team about this last week. Thanks for the heads up.